virtwho/register.py

```diff
@@ -1,5 +1,291 @@
-"""
-class SubscriptionManager()
-class RHSMAPI()
-class SatelliteCLI()
-"""
+import re
+import time
+from virtwho import logger, FailException
+from virtwho.configure import get_register_handler
+from virtwho.ssh import SSHConnect
+
+
+class SubscriptionManager:
+
+    def __init__(self, host, username, password, port=22, register_type='rhsm',
+                 org=None, activation_key=None):
+        """
+        Define the global variables.
+        :param host: ip/hostname to run subscription-manager command.
+        :param username: account username of the host
+        :param password: password to access the host
+        :param port: port to access the host
+        :param register_type: rhsm or satellite, rhsm as default
+        :param org: organization of the entitlement server
+        :param activation_key: activation_key of the entitlement server
+        """
+        self.host = host
+        self.register_type = register_type
+        self.rhsm_conf = '/etc/rhsm/rhsm.conf'
+        self.ssh = SSHConnect(host=host, user=username, pwd=password, port=port)
+        self.registe = get_register_handler(register_type)
```

Comment:
Actually it is not an error because there is register() function blow, so I used the ```self.registe```, but it's really not better, so I changed to ```register```, which is not necessary to define as global para.